plugins/modules/foreman_role.py

```diff
@@ -120,19 +128,29 @@ def main():
 
     filters = entity_dict.pop("filters", None)
 
-    changed, entity = module.ensure_entity('roles', entity_dict, entity)
-
-    if not module.desired_absent and not module.check_mode:
-        if filters is not None:
-            for role_filter in filters:
-                role_filter['role_id'] = entity['id']
-                role_filter['permissions'] = module.find_resources_by_name('permissions', role_filter['permissions'], thin=True)
-                module.ensure_entity_state('filters', role_filter, None, None, 'present', filters_entity_spec)
-            old_filters = entity.get('filter_ids', [])
-            for old_filter in old_filters:
-                module.ensure_entity('filters', None, {'id': old_filter}, {}, 'absent')
-            if len(old_filters) != len(filters):
-                changed = True
+    changed, new_entity = module.ensure_entity('roles', entity_dict, entity)
```

Comment:
I find `entity` vs `new_entity` confusing and had to read multiple times until I understood why you use `entity` below again.

Reply:
Right. It's confusing. But new_entity is the one with an id, and the old entity can tell us whether we started out blank.

Reply:
Blank? Don't we just need the list of filters (if any) at this point?

Reply:
It's in line 138, where we check, whether the entity was newly created before.

Reply:
And `new_entity` doesn't contain `filters` when it's actually an updated entity, so we can't just do `current_filters = [module.show_resource('filters', filter['id']) for filter in new_entity['filters']]`?

Reply:
I think, it was something like that. It's similar with nested parameters.